presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/HistogramStateFactory.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.operator.SingleChannelGroupByHash;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.util.array.ObjectBigArray;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class HistogramStateFactory
+        implements AccumulatorStateFactory<HistogramState>
+{
+    private final Type type;
+    private final int expectedSize;
+
+    public HistogramStateFactory(Type type, int expectedSize)
```

Comment:
This constructor is only used in one place and `expectedSize` is a constant there, so just move that constant to this class and remove the parameter